pyomo/contrib/incidence_analysis/config.py

```diff
@@ -54,7 +65,90 @@ class IncidenceMethod(enum.Enum):
 )
 
 
-IncidenceConfig = ConfigDict()
+class _ReconstructVisitor:
+    pass
+
+
+def _amplrepnvisitor_validator(visitor=_ReconstructVisitor):
+    # This checks for and returns a valid AMPLRepnVisitor, but I don't want
+    # to construct this if we're not using IncidenceMethod.ampl_repn.
+    # It is not necessarily the end of the world if we construct this, however,
+    # as the code should still work.
+    if visitor is _ReconstructVisitor:
+        subexpression_cache = {}
+        subexpression_order = []
+        external_functions = {}
+        var_map = {}
+        used_named_expressions = set()
+        symbolic_solver_labels = False
+        # TODO: Explore potential performance benefit of exporting defined variables.
+        # This likely only shows up if we can preserve the subexpression cache across
+        # multiple constraint expressions.
+        export_defined_variables = False
+        sorter = FileDeterminism_to_SortComponents(FileDeterminism.ORDERED)
+        amplvisitor = AMPLRepnVisitor(
+            text_nl_template,
+            subexpression_cache,
+            subexpression_order,
+            external_functions,
+            var_map,
+            used_named_expressions,
+            symbolic_solver_labels,
+            export_defined_variables,
+            sorter,
+        )
+    elif not isinstance(visitor, AMPLRepnVisitor):
+        raise TypeError(
+            "'visitor' config argument should be an instance of AMPLRepnVisitor"
+        )
+    else:
+        amplvisitor = visitor
+    return amplvisitor
+
+
+_ampl_repn_visitor = ConfigValue(
+    default=_ReconstructVisitor,
+    domain=_amplrepnvisitor_validator,
+    description="Visitor used to generate AMPLRepn of each constraint",
+)
+
+
+class _IncidenceConfigDict(ConfigDict):
+    def __call__(
+        self,
+        value=NOTSET,
+        default=NOTSET,
+        domain=NOTSET,
+        description=NOTSET,
+        doc=NOTSET,
+        visibility=NOTSET,
+        implicit=NOTSET,
+        implicit_domain=NOTSET,
+        preserve_implicit=False,
+    ):
+        init_value = value
+        new = super().__call__(
+            value=value,
+            default=default,
+            domain=domain,
+            description=description,
+            doc=doc,
+            visibility=visibility,
+            implicit=implicit,
+            implicit_domain=implicit_domain,
+            preserve_implicit=preserve_implicit,
+        )
+
+        if (
+            new.method == IncidenceMethod.ampl_repn
+            and "ampl_repn_visitor" not in init_value
+        ):
+            new.ampl_repn_visitor = _ReconstructVisitor
```

Comment:
If we are using `ampl_repn` and did not provide a visitor, explicitly trigger reconstruction of the visitor.